cdap-ui/app/cdap/components/AbstractWidget/InputFieldDropdown/index.tsx

```diff
@@ -52,14 +62,53 @@ const InputFieldDropdown: React.FC<IInputFieldProps> = ({
   onChange,
   disabled,
   extraConfig,
+  widgetProps,
 }) => {
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
-  const fieldValues = getFields(inputSchema);
-  const widgetProps = {
-    options: fieldValues,
-  };
 
-  return <Select value={value} onChange={onChange} widgetProps={widgetProps} disabled={disabled} />;
+  const isMultiSelect: boolean = widgetProps.multiselect || false;
+  const allowedTypes: string[] = widgetProps.allowedTypes || [];
+
+  const fieldValues = getFields(inputSchema, allowedTypes);
+
+  const newValue = value
+    .toString()
+    .split(delimiter)
+    .filter((v) => fieldValues.includes(v))
+    .toString();
+
+  if (newValue !== value) {
+    onChange(newValue);
+    value = newValue;
```

Comment:
`value` is prop coming from parent component, you shouldn't do this. `onChange(newValue)` would update the prop `value` for you

Reply:
By "shouldnt do this" are you referring to the whole if statement or explicitly updating value? I was concerned that always calling `onChange()` would result in an infinite loop.

Reply:
explicitly updating the prop `value`.

Reply:
Done